### PR TITLE
Add support for PR comments

### PR DESCRIPTION
**Pull request by Pradoxzon**
Adding support for working with PR comments.

The purpose of this feature is to lay some groundwork for allowing Policy as Code to create PR comment summaries of workflow runs (see [Issue 42 in PaC](https://github.com/advanced-security/policy-as-code/issues/42) for reference).

Open to suggestions for changes or different implementations!

### Get PR comments
- REST API - `/repos/{owner}/{repo}/issues/{issue_number}/comments`
- GET
- `issue_number` - the PR #

### Create new PR comment
- REST API - `/repos/{owner}/{repo}/issues/{issue_number}/comments`
- POST
- `issue_number` - the PR #
- body in request - string representing the comment to be created
- expected response code - 201

### Update existing PR comment
- REST API - `/repos/{owner}/{repo}/issues/comments/{comment_id}`
- PATCH
- `comment_id` - ID of the existing comment to update
- body in request - string representing the updated comment contents, this replaces the previous contents of the comment
- expected response code - 200

### CODE REVIEW

**Comment by GeekMasher** [comment]
@Pradoxzon can you format the code for me?

File: src/ghastoolkit/octokit/github.py [hunk 1]
```diff
@@ -99,6 +99,53 @@ def getPullRequestCommits(self) -> list[str]:
                 result.append(commit.get("sha"))
         return result
 
+    def getPullRequestComments(self) -> list[dict[str, int | str]]:
```
Comment (GeekMasher) [inline src/ghastoolkit/octokit/github.py:102]:
```suggestion
    def getPullRequestComments(self) -> list[dict[str, Union[int, str]]]:
```

To support Python v3.9 you will need to use `Union` type

Comment (Pradoxzon) [reply src/ghastoolkit/octokit/github.py]: Missed that one, thanks!

Comment (GeekMasher) [reply src/ghastoolkit/octokit/github.py]: Might need to import the type too

```python
from typing import Optional, Tuple, Union
```

**Comment by GeekMasher** [comment]
Might need to import the type too

```python
from typing import Optional, Tuple, Union
```

**Comment by GeekMasher** [comment]
@Pradoxzon can you pull in main? I should have fixed the Action step, if not I'll merge it and fix Actions later

**Comment by GeekMasher** [comment]
Awesome, once CodeQL is finished I'll merge it

**Pull request merged**